lib/serverengine/daemon_logger.rb

```diff
@@ -55,6 +55,7 @@ def logdev=(logdev)
         # update path string
         old_file_dev = @file_dev
         @file_dev = LogDevice.new(logdev, shift_age: @rotate_age, shift_size: @rotate_size)
+        @file_dev.extend(RotationAware) if @rotate_age == 0 && !ServerEngine.windows?
         old_file_dev.close if old_file_dev
```

Comment:
If log_rotate_age was not set, log_rotate_age: 5(default) is used.
Does it work intentionally in such a case?

Reply:
This change intend to reopen the log file only when the application doesn't rotate the file by itself.
(When such case, the application expects that the log is rotated by external tool.)

When `log_rotate_age` is non-zero (including `nil`), the log is rotated by the application.
Please see also the implementation of `LogDevice` of Ruby:

https://github.com/ruby/ruby/blob/abc828bc67ff42ddb42f5bd034cec185d6b8821d/lib/logger/log_device.rb#L14-L28
https://github.com/ruby/ruby/blob/abc828bc67ff42ddb42f5bd034cec185d6b8821d/lib/logger/log_device.rb#L123-L136

Fluentd sets `log_rotate_age` to `0` by default.

Reply:
> Fluentd sets `log_rotate_age` to `0` by default.

Sorry, it's wrong, it sets `nil` by default.

Reply:
I've overlooked that situation, thanks.
So go forward this PR.

Reply:
Or it would be better that we always enable it, so that users can delete log file manually to recover free storage space.

Reply:
I'll remove the `if` statement.

Reply:
Removed in 88420811ac563225a5420371b295968f99f9cfa5